File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    answer = []
    box = []
    for i in range(2, n+1):
        if n % i == 0:                     # 약수인가?
            for j in range(2, i):
                if i % j == 0:             # 소수인가?
                    box.append(j)
            if len(box) == 0:
                answer.append(i)
        box = []                           # 박스가 바깥 for과 같아지면
    return answer

File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collector;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;
import java.util.Scanner;
class Solution {
    public int[] solution(int n) {
    List <Integer> answer = new ArrayList<>();
    int[] net = new int[n+1];
    for (int i = 2 ; i < n+1; i++){
        if (n % i == 0 && net[i] == 0){
            answer.add(i);
            for (int j = i; j < n+1; j += i){
                net[j] = 1;
            }
        }
    }
    return answer.stream().mapToInt(Integer::intValue).toArray();
}


}